### ISSUE TITLE
Upgrade parser for new syntax support

### ISSUE THREAD

**Issue by motiz88** [flowtype/flow-remove-types#89]
This is a master issue tracking the replacement of the Babel v6 parser with the latest Flow parser, enabling full support for the latest syntax Flow understands.

Among other things, this will include support for:

* Explicit type arguments in calls
* JSX fragments

**Comment by micahstubbs** [comment]
noticed the issue with JSX fragments as well

```
 ↳ Syntax Error: Unexpected token (113:11)
   112:            <>
error Command failed with exit code 1.
```

![Screen Shot 2019-04-30 at 3 10 07 PM](https://user-images.githubusercontent.com/2119400/56996677-2005f200-6b5a-11e9-985b-eded289474cf.png)

**Comment by motiz88** [comment]
We've just published v2.99.0 of both `flow-remove-types` and `flow-node` containing this change. Future releases will be aligned with Flow releases (e.g. v2.99.0 ↔ Flow 0.99.0) and will always use the latest version of the Flow parser for maximum compatibility.

The code has now [moved](https://github.com/facebook/flow/tree/master/packages/flow-remove-types) to the main Flow repo, and this separate repo will be archived soon; please use the Flow issue tracker for any further discussion.
